tests/helpers.el

```diff
@@ -0,0 +1,35 @@
+;;; -*- lexical-binding: t; -*-
+
+(defvar *test-data-root* (expand-file-name
+                          "data/"
+                          (file-name-directory load-file-name)))
+
+(defvar *test-temp-dir*)
+(defvar *test-git-repo*)
+
+(defun set-up-temp-dir ()
+  (setf *test-temp-dir* (make-temp-file "magit-p4-test" t)
+        default-directory *test-temp-dir*))
+
+(defun tear-down-temp-dir ()
+    (delete-directory *test-temp-dir* t))
+
+(defun set-up-git-repo ()
+  (let ((src (expand-file-name "_git" *test-data-root*))
+        (target (expand-file-name "magit-p4/.git" *test-temp-dir*)))
+    (copy-directory src target nil t t)
+    (setf *test-git-repo* (file-name-directory target))))
+
+(defun buffer-contains-p (string &optional buffer)
+  (let ((buffer (or buffer (current-buffer))))
+    (with-current-buffer buffer
+      (save-excursion
+        (goto-char (point-min))
+        (search-forward string nil t)))))
+
+(defun simulate-keys (keys)
+  (setf unread-command-events (append unread-command-events
+                                      (listify-key-sequence (kbd keys)))))
+
+
+(provide 'helpers)
```

Comment:
This is a test utility, not a library intended for end-users and expected to follow conventions such as namespaced symbol prefixes. So it shouldn't provide a feature, and it should be loaded using `load`. Maybe you can use a relative path or maybe you will have to involve `(or load-file-name buffer-file-name)` in the path construction.

While that isn't strictly necessary, as long as you don't provide a feature, I think the file-name should be namespaced. "helper" is a reasonable name, but anyone else may chose the same name.

Reply:
Is this really a concern though? The file is already located under `tests/`, so it's not going to be in the `load-path` in the first place, will it?

Reply:
The tool I use extract metadata for the Emacsmirror would find it and add `helpers` to the features provided by this package. Another way to combat that is to add `tests/.nosearch`. It does this because some package repositories have libraries in more than one location (e.g., the top-level as well as `extensions/`).

Reply:
Either way, done.

Reply:
Ah, that makes sense, I didn't think of other code scanning it.